### PR TITLE
expose xor gate

### PR DESCRIPTION
**Pull request by Trivo25**
o1js https://github.com/o1-labs/o1js/pull/1177

### CODE REVIEW

File: crypto/finite_field.ts [hunk 1]
```diff
@@ -123,6 +123,9 @@ function createField(p: bigint, t: bigint, twoadicRoot: bigint) {
     t,
     twoadicRoot,
 
+    xor(x: bigint, y: bigint) {
+      return y ^ x;
+    },
```
Comment (mitschabaude) [inline crypto/finite_field.ts:128]: what I don't love about this function is that the output is not necessarily another field element (i.e. smaller than the modulus)

it seems a bit out of place in a finite field library. maybe we need a bigint library as well. maybe for another PR.

File: ocaml/lib/snarky_bindings.ml [hunk 2]
```diff
@@ -141,6 +141,43 @@ module Gates = struct
...
+                   ; in2_3
+                   ; out_0
+                   ; out_1
+                   ; out_2
+                   ; out_3
+                   } )
+          } )
+
+  let zero_check in1 in2 out =
+    Impl.with_label "zero_check" (fun () ->
```
Comment (mitschabaude) [inline ocaml/lib/snarky_bindings.ml:172]: as discussed in the other PR this should probably just be called `zero`

**Review by mitschabaude — APPROVED** [review]
pre-approving with a nit

File: crypto/finite_field.ts (hunk 1 above)
Comment (mitschabaude) [reply crypto/finite_field.ts]: sorry for being annoying but I don't like that either 😅 that behavior is unexpected. it's fine to leave the function here for now but better remove the `mod()`

Comment (Trivo25) [reply crypto/finite_field.ts]: That's fine, I don't mind at all! I agree its an odd place for it to live in :D

**Pull request merged**